lib/test_ids/allocator.rb

```diff
@@ -98,21 +146,21 @@ def allocate(instance, options)
       end
 
       # Otherwise generate the missing ones
-      bin['number'] ||= allocate_bin(size: bin_size)
+      bin['number'] ||= allocate_bin(options.merge(size: bin_size))
       bin['size'] ||= bin_size
       # If the softbin is based on the test number, then need to calculate the
       # test number first.
       # Also do the number first if the softbin is a callback and the number is not.
       if (config.softbins.algorithm && config.softbins.algorithm.to_s =~ /n/) ||
          (config.softbins.callback && !config.numbers.function?)
-        number['number'] ||= allocate_number(bin: bin['number'], size: number_size)
+        number['number'] ||= allocate_number(options.merge(bin: bin['number'], size: number_size))
         number['size'] ||= number_size
-        softbin['number'] ||= allocate_softbin(bin: bin['number'], number: number['number'], size: softbin_size)
+        softbin['number'] ||= allocate_softbin(options.merge(bin: bin['number'], number: number['number'], size: softbin_size))
         softbin['size'] ||= softbin_size
       else
-        softbin['number'] ||= allocate_softbin(bin: bin['number'], size: softbin_size)
+        softbin['number'] ||= allocate_softbin(options.merge(bin: bin['number'], size: softbin_size))
         softbin['size'] ||= softbin_size
-        number['number'] ||= allocate_number(bin: bin['number'], softbin: softbin['number'], size: number_size)
+        number['number'] ||= allocate_number(options.merge(bin: bin['number'], softbin: softbin['number'], size: number_size))
```

Comment:
Thanks for adding the options to all of these method calls, but the feature is only half-implemented if the corresponding calls to the callbacks don't pass the options - i.e. you have done it for the softbin callback you care about, why not the rest?